src/components/elements/ingredient/FormIngredient.tsx

```diff
@@ -0,0 +1,139 @@
+    marginTop: theme.spacing(3),
+  },
+  textField: {
+    width: '10ch',
+  },
+}));
+
+
+type Ingredient = {
+    name: string,
+    inventory: number,
+    unit: string,
+    price: number,
+}
+
+type FormIngredientProps = {
+    nextStep: (step:number) => void,
+    forStep:number,
+    backStep:number,
+    handleIngredient: (item: Ingredient) => void,
+    ingredientItemFromSelect: Ingredient,
+
+}
+
+
+export const FormIngredient = (props: React.PropsWithChildren<FormIngredientProps>) => {
+  const {nextStep, forStep, backStep, handleIngredient, ingredientItemFromSelect} = props;
+  const classes = useStyles();
+
+  const [ingredientItem, setIngredientItem] = React.useState<Ingredient>(ingredientItemFromSelect);
+
+  const updateIngredientItem = (key: string) => (e: ChangeEvent<HTMLTextAreaElement>) => {
+    setIngredientItem({...ingredientItem, [key]: e.target.value});
+    handleIngredient(ingredientItem);
+  };
+  const updateIngredientNum = (key: string) => (e: ChangeEvent<HTMLInputElement>) => {
+    setIngredientItem({...ingredientItem, [key]: +e.target.value});
+    handleIngredient(ingredientItem);
+  };
+
+  return (
+    <React.Fragment>
+      <FormControl fullWidth={false} className={classes.margin} variant="filled">
+        <h3>Fill in Ingredient Item Things and change this line</h3>
+        <TextField
+          id="Name"
+          label="Name"
+          onChange={updateIngredientItem('name')}
+          defaultValue={ingredientItem.name}
+          variant="filled"
+        />
+        <br/>
+        <TextField
+          id="Unit"
+          label="Unit of Measurement for Ingredient"
+          onChange={updateIngredientItem('unit')}
+          defaultValue={ingredientItem.unit}
+          variant="filled"
+        />
+        <br/>
+        <FormControl fullWidth = {false} className={classes.margin} variant="filled">
+          <InputLabel htmlFor="filled-adornment-password">
+                Inventory - Current Available Units of Ingredient
+          </InputLabel>
+          <FilledInput
+            type = 'number'
+            onChange={updateIngredientNum('inventory')}
+            defaultValue={ingredientItem.inventory}
+            id="filled-adornment-password"
+          />
+        </FormControl>
+        <br/>
+        <FormControl fullWidth = {false} className={classes.margin} variant="filled">
+          <InputLabel htmlFor="filled-adornment-password">
+                Price - Price to buy One Unit Worth of Inventory
+          </InputLabel>
+          <FilledInput
+            type = 'number'
+            onChange={updateIngredientNum('price')}
+            defaultValue={ingredientItem.price}
+            id="filled-adornment-password"
+            startAdornment={<InputAdornment position="start">$</InputAdornment>}
+          />
+        </FormControl>
+        <Button
+          variant='contained'
+          color='primary'
+          style={styles.button}
+          onClick={function() {
```

Comment:
Make this a function above and call it here. No need to have all this logic here.